Added Skill Tree

mendozatg opened the pull request:
Added Skill Tree for respective classes and buffs, will edit if requested

For Classes: Alchemist, Assassin, Healer, Knight, Mage, Ranger, Wizard
- Unique Names for the skill tree branches
- Methods: Increases specific base stats

pacalso-jjc commented on wiki/Class-Diagram.md:
These are classes for the different available character class, I was looking for just the class for Skill tree which might contain the following
List of available skills
Unlocked skills
Skill points

pacalso-jjc commented:
There is an error during rendering the image
![image](https://github.com/user-attachments/assets/15a7dc20-f97d-4204-8be1-82fe5926c045)

pacalso-jjc reviewed (changes requested):
Kindly check rendering it is showing like this to my end
![image](https://github.com/user-attachments/assets/a3108505-7a40-4e99-9c7b-b5ed8509588b)

mendozatg commented:
Sir I cant view it properly since the image is blurry but on my end, it is only showing this part
[
![image_2025-03-16_092814592](https://github.com/user-attachments/assets/06cd5e70-380e-4587-96d8-bd2c95c5d30f)
](url)

mendozatg commented:
Sir I changed my changes, if that makes sense. I fixed some syntax errors in my branch and it should show if it is merged. Please let me know sir if I should fix anything else.

pacalso-jjc reviewed (changes requested):
Please fix other review points before I approve

pacalso-jjc commented on wiki/Class-Diagram.md:
```diff
@@ -59,6 +58,12 @@ classDiagram
     }
 
+    class SkillTree {
+        - int skillPoints
```
This might also need to have an attribute for skillList of type List<Skill>
